### DISCUSSION TITLE
Hikvision Camera Stream setup

### DISCUSSION THREAD

**Discussion by travismailsa** [blakeblackshear/frigate#9556] [Ask A Question]
Hi,

I am new to Frigate  and configuration, but i cannot find clear documentation as  how to setup/config a single Hikvision camera to the frigate.yml file. If i remove my camera stream setup, Frigate starts. If i add the setup - what i have managed to find through videos, it crashes. No clue where to start - help would be much appreciated.

"cameras:
  ...
      ffmpeg:
      inputs:
        - path: rtsp://xxx:xxxx@192.168.1.112:554/Streaming/Channels/101
          roles:
            - detect
            - record
    rtmp:
      enabled: True
    record:
      enabled: True
    snapshots:
      enabled: True
    detect:
      enabled: True
      width: 1280
      height: 720
      fps: 5
    objects:
      track:
        - person
        - cars
        "
RTSP url is correct and working
![image](https://github.com/blakeblackshear/frigate/assets/68334369/6560c641-6263-43cb-b4d9-b5d76d6e5827)

**Comment by schford** [comment]
Mine below for one hikvision camera if that helps.  Don't think you need that port in url.

go2rtc:
  streams:
    garage:
    - rtsp://admin:password@192.168.100.44/Streaming/Channels/101
    garage_sub:
    - rtsp://admin:password@192.168.100.44/Streaming/channels/102 # 6frames

  garage_camera:
    ffmpeg:
      hwaccel_args: preset-vaapi # Use Intel NUC iGPU (optional)
      inputs:
      - path: rtsp://127.0.0.1:8554/garage_sub
        input_args: preset-rtsp-restream
        roles:
        - detect
      - path: rtsp://127.0.0.1:8554/garage
        input_args: preset-rtsp-restream
        roles:
        - record
    detect:
      enabled: true # <---- disable detection until you have a working camera feed
      width: 640 # <---- update for your camera's resolution
      height: 480 # <---- update for your camera's resolution
      fps: 6
    objects:
      track:
      - person
      - dog
      - cat
      - car
      - bicycle
    motion:
      mask:
      - 145,37,217,41,217,0,143,0
    snapshots:
      enabled: true
      required_zones:
      - drive_steps
    record:
      enabled: true
      events:
        required_zones:
        - drive_steps
        retain:
          default: 10
    live:
      stream_name: garage

    zones:
      drive_steps:
        coordinates: 248,165,192,0,0,0,0,480,431,480,410,437,613,303,593,91

**Reply by travismailsa** [reply]
Thanks for the reply and your setup, i will give it a go this evening and update here

**Reply by travismailsa** [reply]
I have made progress, any idea why when i start adding more cameras i get the below on the image? Worked for a good few hours, added a camera and then it started. It did the same when i added the second camera? My PTZ is a hikvision, other is a 180 degree hik and lastly the other is a standard 4mp hik bullet - i have more to add, but doing it in phases.
![image](https://github.com/blakeblackshear/frigate/assets/68334369/dea1411d-ea86-4a25-8420-904218952aea)

**Reply by NickM-27** [reply]
Have you enabled hwaccel?

**Reply by travismailsa** [reply]
Now 180 is working after restarting Frigate
![image](https://github.com/blakeblackshear/frigate/assets/68334369/0321b00c-3259-4d6b-9eee-ce12611e627f)

Removed the 180

![image](https://github.com/blakeblackshear/frigate/assets/68334369/52bc3911-3f8e-4b04-90f8-454bbc663574)

Now entrance works - using ffmpeg setup with RSTP high res for recording and low res for detect

**Reply by travismailsa** [reply]
![image](https://github.com/blakeblackshear/frigate/assets/68334369/e02632c1-5495-450e-b382-89957ed69a28)
Events working fine when i returned home on my bike, but when i add more cameras it seems to go into a wobble.
Running it on a i7 32 gig ram server - no gpu/tpu using cpu - forgot to mention i am running ESXI VM of HA

**Reply by NickM-27** [reply]
Then that's the problem, your device isn't able to keep up. Using GPU for hwaccel and GPU or coral for object detection is highly recommended.

**Reply by travismailsa** [reply]
I figured as much tbh, sadly coral isnt readily available in South Africa

**Reply by travismailsa** [reply]
Can i not at more virtual cores instead of just setting it to cpu0 ? Because as far as i am aware the VM wouldnt recoginse them as threads for the added command?

**Reply by travismailsa** [reply]
![image](https://github.com/blakeblackshear/frigate/assets/68334369/689b0ba7-7f79-4bbf-9ae9-1e1df6e9dc33)
Add more here?

detectors:
  cpu0:
    type: cpu

**Reply by NickM-27** [reply]
You can add more cpu detectors

**Reply by travismailsa** [reply]
@NickM-27 , 100% right, maxing out my cpu shown below on ESZXI -shite
![image](https://github.com/blakeblackshear/frigate/assets/68334369/8e217e98-2643-4c7d-ae01-a416a53889f7)

**Reply by schford** [reply]
https://www.desertcart.co.za/products/125589101-google-coral-usb-edge-tpu-ml-accelerator-coprocessor-for-raspberry-pi-and-other-embedded-single-board-computers

**Comment by travismailsa** [comment]
> https://www.desertcart.co.za/products/125589101-google-coral-usb-edge-tpu-ml-accelerator-coprocessor-for-raspberry-pi-and-other-embedded-single-board-computers

Much aprreciaed and thanked, but desertcart.co.za is a dropshipping site from Aliexpress - never know what you are going to get or if it will ever arrive in the country .

Think ill buy the unit from a European reputable supplier i think  and pay the hefty custom fees using DHL and FEDEK etc- ive seen its only around 70$..

I might move my HA over to my gaming PC and run a test with my RTX 3080

**Reply by schford** [reply]
the coral works super super well if you do get the chance to lay your hands on one.

Cheers

Stuart

**Reply by travismailsa** [reply]
Thanks a lot mate for taking the time to reply. 
If my thread does not get closed i will post the results - whether it solved the issue or now
Cheers
Travis

**Comment by Cold-Lemonade** [comment]
@travismailsa I am trying to set up a Anpviz camera, but Frigate's ffmpeg keeps dropping it. I believe the Anpviz camera is the same (more or less) to Hikvision. Would you be so kind as to share your ```config.yml``` code that you used to configure your Hikvision camera? I'd like to see how you did it. Thanks.
